core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1223,7 +1247,7 @@ class KafkaApis(val requestChannel: RequestChannel,
         Set.empty[MetadataResponseTopic]
       else
         unauthorizedForDescribeTopics.map(topic =>
-          metadataResponseTopic(Errors.TOPIC_AUTHORIZATION_FAILED, topic, false, util.Collections.emptyList()))
+          metadataResponseTopic(Errors.TOPIC_AUTHORIZATION_FAILED, topic, metadataCache.getTopicId(topic), false, util.Collections.emptyList()))
```

Comment:
Same here.

Reply:
We should check some of the other error responses to make sure we use name/id appropriately

Reply:
I checked the error responses and found that we should hide TopicId when error=`TOPIC_AUTHORIZATION_FAILED`, and we can respond topicId on other errors.

Reply:
Thanks for fixing this!